packages/sdk/src/modules/did/did.spec.ts

```diff
@@ -18,72 +20,110 @@ describe('Did', () => {
     keyring = new Keyring({ type: 'sr25519' });
     alice = keyring.addFromUri('//Alice');
     did = new Did(api, { pair: alice });
-  });
+  }, 40000);
 
   afterAll(async () => {
     await unsubscribeRuntimeVersion(api);
     await api?.disconnect();
   });
 
   describe('create()', () => {
-    it('should create a new Did', async () => {
-      const name = `test-did-${generateRandomString()}`;
+    // creates, reads, and removes to align with expected
+    it('create single did with no custom fields', async () => {
+      const new_did  = 'did-test-1';
+      await createReadRemove(new_did, did, alice, null);
+    }, 80000);
+
+    // try to create a did of the same name -> expect error
+    it('create did of same name error', async () => {
+      const new_did  = 'did-test-1';
+      await did.create({name: new_did});
+
+      await expect(did.create({ name: new_did }))
+        .rejects.toThrow("Create DID Error: AttributeAlreadyExist for peaqDid.");
+
+      // remove did for cleanup
+      const removeResult = await did.remove({name: new_did});
+      expect(removeResult?.hash).toBeDefined();
+      expect(typeof removeResult?.unsubscribe).toBe('function');
+    }, 80000);
+
+    // did with service custom field
+    it('create custom did with service custom field', async () => {
+      const new_did  = 'did-test-1';
+
+      // create proto document to represent new custom service did documents fields
+      const peaqServiceProto = new peaqDidProto.Service();
```

Comment:
using peaqProto to build the customDocumentFields. Change customDocumentFields in the did/index.ts to be expecting these types.